QuizApp/Screens/QuestionPage/QuizViewController.swift

```diff
@@ -140,77 +173,153 @@ final class QuizViewController: UIViewController {
-        
+        viewModel.moveToNextQuestion()
     }
 }
 
 // MARK: - UITableViewDataSource
 extension QuizViewController: UITableViewDataSource {
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return answers.count
+        return viewModel.numberOfOptions
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        guard let cell = tableView.dequeueReusableCell(withIdentifier: AnswerCell.identifier, for: indexPath) as? AnswerCell else {
+        guard let cell = tableView.dequeueReusableCell(
+            withIdentifier: AnswerCell.identifier,
+            for: indexPath
+        ) as? AnswerCell else {
             return UITableViewCell()
         }
         
-        cell.configure(with: answers[indexPath.row])
+        cell.configure(with: viewModel.getOptionTitle(at: indexPath.row))
         
-        if let selectedIndex = selectedAnswerIndex {
+        if let selectedIndex = viewModel.selectedAnswerIndex {
             if selectedIndex == indexPath.row {
-                let isCorrect = selectedIndex == correctAnswerIndex
-                cell.updateState(isCorrect: isCorrect)
-            } else if indexPath.row == correctAnswerIndex {
+                cell.updateState(isCorrect: viewModel.isOptionCorrect(at: indexPath.row))
+            } else if viewModel.isOptionCorrect(at: indexPath.row) {
                 cell.markAsCorrect()
             } else {
                 cell.resetState()
             }
         } else {
             cell.resetState()
         }
-        
         return cell
     }
-    
-    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
-        return Constants.QuizViewControllerConstants.heightForRow
-    }
 }
 
 // MARK: - UITableViewDelegate
 extension QuizViewController: UITableViewDelegate {
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        guard selectedAnswerIndex == nil else { return }
-        
-        selectedAnswerIndex = indexPath.row
-        let isCorrect = indexPath.row == correctAnswerIndex
-        
-        if let selectedCell = tableView.cellForRow(at: indexPath) as? AnswerCell {
-            selectedCell.updateState(isCorrect: isCorrect)
-        }
-        
-        if !isCorrect {
-            let correctIndexPath = IndexPath(row: correctAnswerIndex, section: 0)
-            if let correctCell = tableView.cellForRow(at: correctIndexPath) as? AnswerCell {
-                correctCell.markAsCorrect()
-            }
-        }
-        tableView.allowsSelection = false
+        viewModel.selectAnswer(at: indexPath.row)
+    }
+}
+
+// MARK: - QuizViewModelDelegate
+extension QuizViewController: QuizViewModelDelegate {
+    func didUpdateQuestion() {
+        answersTableView.allowsSelection = true
+        configureViews()
+    }
+    
+    func didCompleteQuiz(withScore points: Int) {
```

Comment:
I think you are not using points arguments. do you really need this function to have it?

Reply:
i am using it in view model completeQuiz method